CI issue: we don't have rust-src on CI

```
error: "/home/runner/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/Cargo.lock" does not exist, unable to build with the standard library, try:
        rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu
Error: Process completed with exit code 101.
```

can we install the said component on CI?

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -38,8 +38,8 @@ jobs:
           target: ${{ matrix.target }}
           override: true
           components: rustfmt, clippy
+    - run: rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu
     - name: Build
       run: cargo build --verbose
     - name: Lint
       run: cargo clippy --verbose
-    - run: rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -38,7 +38,6 @@ jobs:
           target: ${{ matrix.target }}
           override: true
           components: rustfmt, clippy
-    - run: rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu
     - name: Build
       run: cargo build --verbose
     - name: Lint

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -42,3 +42,4 @@ jobs:
       run: cargo build --verbose
     - name: Lint
       run: cargo clippy --verbose
+    - run: rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -38,6 +38,8 @@ jobs:
           target: ${{ matrix.target }}
           override: true
           components: rustfmt, clippy
+    - name: Add nightly-x86_64
+      run: rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu
     - name: Build
       run: cargo build --verbose
     - name: Lint

